org.eclipse.xtext/src/org/eclipse/xtext/nodemodel/INode.java

```diff
@@ -156,6 +156,7 @@
 	/**
 	 * Returns the parsed text that is covered by this node (including hidden tokens). The result is never <code>null</code>
 	 * but may be empty.
+	 * @throws IllegalStateException if text can't be computed because rootNode is 'null'.
```

Comment:
I'd rephrase this to `if the text is not available. A possible reason is, that the node is not part of a valid node model`